controllers/topology/desired_state.go

```diff
@@ -229,9 +229,25 @@ func computeControlPlaneVersion(s *scope.Scope) (string, error) {
 		return "", errors.Wrap(err, "failed to get the version from control plane spec")
 	}
 
-	// Return early if the version is already equal to the desired version
-	// no further checks required.
+	s.UpgradeTracker.ControlPlane.PendingUpgrade = true
 	if *currentVersion == desiredVersion {
+		// Mark that the control plane spec is already at the desired version.
```

Comment:
This block could be simplified to: (if you want)
```go
s.UpgradeTracker.ControlPlane.PendingUpgrade = *currentVersion != desiredVersion
```

Reply:
I am going to leave it as is. Provides a good place to add a comment block to explain what/why we are doing this step.

Reply:
Absolutely fine, makes sense.